libraries/lib-wave-track/SampleBlock.h

```diff
@@ -44,6 +45,8 @@ class MinMaxRMS
 class WAVE_TRACK_API SampleBlock
 {
 public:
+   virtual LibSampleTrack::BlockPtr Get() = 0;
```

Comment:
I would prefer a solution that does not require “polluting” the interface of SampleTrack with any mention of caching, but instead, lift the creation of caching scope up to places where WaveTrack is known, and we also construct the “decorator” tracks.

And that must wait on my other refactoring work, part of which I asked for review yesterday.

Reply:
I misplaced my comment— modifying this class in lib-wave-track is acceptable.  But the new types it uses should also be defined in this library and not in the lower one.